csharp/extractor/Semmle.Extraction.CIL/Entities/NoMetadataHandleType.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Semmle.Util;
+
+namespace Semmle.Extraction.CIL.Entities
+{
+    internal sealed partial class NoMetadataHandleType : Type
+    {
+        private readonly string originalName;
+        private readonly string name;
+        private readonly string? assemblyName;
+        private readonly string containerName;
+        private readonly bool isContainerNamespace;
+
+        private readonly Lazy<TypeTypeParameter[]>? typeParams;
```

Comment:
Why `Lazy`?

Reply:
Just to keep the symmetry with `TypeReferenceType`.